vocode/streaming/agent/factory.py

```diff
@@ -50,3 +52,8 @@ def create_agent(
         elif isinstance(agent_config, MistralAgentConfig):
             return MistralAgent(agent_config=agent_config, logger=logger)
         raise Exception("Invalid agent config", agent_config.type)
+
+    async def new_agent(
```

Comment:
nit: since `new_agent` sounds very similar to `create_agent`, can we rename this to something more descriptive? maybe something like `get_cached_agent`